diff --git a/lib/agents.js b/lib/agents.js
index 8514f9892bb0c04b73689e6ff87447b90e5597eb..a8c2009b9533913a5d95dbe31da8114d4203c2a9 100644
--- a/lib/agents.js
+++ b/lib/agents.js
@@ -32,7 +32,7 @@ module.exports = class Agent extends AgentBase {
   }
 
   get proxy () {
-    return this.#proxy ? { url: this.#proxy } : {}
+    return this.#proxy ? { url: this.#proxy.toString() } : {}
   }
 
   #getProxy (options) {
